### PR TITLE
Introduce Protocol Annotations

### PR DESCRIPTION
**Pull request by erik-3milabs**
This PR annotates the code in this crate with the variable names used to represent the same concepts in the "[2PC-MPC: Emulating Two Party ECDSA in Large-Scale MPC (v20240217)](https://eprint.iacr.org/archive/2024/253/20240217:153208)" paper.

To enhance the readability of the code, additionally, the order of some object instantiations was rearranged. Care was taken to ensure these rearrangements do not introduce flaws in the protocol execution. Still, this PR should be carefully examined to verify this claim.

### CODE REVIEW

**Review by ycscaly — APPROVED** [review]
This will be very helpful for readability & audability!

**Review by ofirfriedman42 — COMMENTED** [review]
very nice

**Review by ofirfriedman42 — APPROVED** [review]

File: src/dkg/centralized_party/commitment_round.rs [hunk 1]
```diff
@@ -78,6 +78,9 @@ impl<
         ProtocolContext,
     >
 {
+    /// This function implements Protocol 4, step 1 of the
+    /// 2PC-MPC: Emulating Two Party ECDSA in Large-Scale MPC paper.
+    /// src: https://eprint.iacr.org/2024/253
```
Comment (dolevmu) [inline src/dkg/centralized_party/commitment_round.rs:83]:
```suggestion
    /// src: https://eprint.iacr.org/archive/2024/253/20240217:153208
```
ePrint keeps track of the version of submission, the numbering might change.

File: src/dkg/centralized_party/commitment_round.rs [hunk 2]
```diff
@@ -78,6 +78,9 @@ impl<
         ProtocolContext,
     >
 {
+    /// This function implements Protocol 4, step 1 of the
+    /// 2PC-MPC: Emulating Two Party ECDSA in Large-Scale MPC paper.
```
Comment (dolevmu) [inline src/dkg/centralized_party/commitment_round.rs:82]:
```suggestion
    /// This function implements step 1 of Protocol 4 (DKG).
    /// 2PC-MPC: Emulating Two Party ECDSA in Large-Scale MPC paper.
```

Comment (dolevmu) [reply src/dkg/centralized_party/commitment_round.rs:82]: 1. Current phrasing sounds like Protocol 4 is the first step of the 2PC-MPC (which is kind of true, its the DKG).
2. Second sentence probably doesn't belong in here.
3. I would add what this step does: samples user share x_A, computes X_A + zk-proof and commits to it.

**Review by dolevmu — APPROVED** [review]
Great 👍 
First comments apply to everything

File: src/dkg/centralized_party/commitment_round.rs (hunk 1 above)
Comment (erik-3milabs) [reply src/dkg/centralized_party/commitment_round.rs]: Applied this across the board.

File: src/dkg/centralized_party/commitment_round.rs (hunk 2 above)
Comment (erik-3milabs) [reply src/dkg/centralized_party/commitment_round.rs]: Applied this across the board.

**Review by dolevmu — APPROVED** [review]

**Pull request merged**